File: CSharp/1179 - Preenchimento de Vetor IV.cs
using System;

class URI {

  static void ImprimirArray (int [] A, int contador, string tipo) {
    for (int i = 0; i < contador; i++) {
      Console.WriteLine("{0}[{1}] = {2}", tipo, i, A[i]);
    }
  }
  
  public static void Main (string[] args) {

    int contPar = 0, contImpar = 0, X = 0;
    int [] impar = new int [15], par = new int [15];

    for (int i = 0; i < 15; i++) {
      X = int.Parse(Console.ReadLine());
      if (X % 2 == 0) {
        par[contPar] = X;
        contPar ++;
      }
      else {
        impar[contImpar] = X;
        contImpar ++;
      }
      if (contPar == 5) {
        ImprimirArray(par, contPar, "par");
        contPar = 0;
      }
        
      if (contImpar == 5){
        ImprimirArray(impar, contImpar, "impar");
        contImpar = 0;
      }
        
    }

    if (contImpar > 0) {
      ImprimirArray(impar, contImpar, "impar");
    }
    if (contPar > 0)
      ImprimirArray(par, contPar, "par");
    
    


  }
}

File: Python 3.8/1179 - Preenchimento de Vetor IV.py
IMPAR = []
PAR = []
for i in range(15):
    X = int(input())
    if X % 2 == 0:
        PAR.append(X)
    elif X % 2 != 0:
        IMPAR.append(X)
    if len(PAR) == 5:
        Y = 0
        for j in PAR:
            print('par[{}] = {}'.format(Y,j))
            Y += 1
        PAR = []
    if len(IMPAR) == 5:
        Y = 0
        for j in IMPAR:
            print('impar[{}] = {}'.format(Y,j))
            Y += 1
        IMPAR = []
if len(IMPAR) > 0:
    Y = 0
    for j in IMPAR:
        print('impar[{}] = {}'.format(Y,j))
        Y += 1
if len(PAR) > 0:
    Y = 0
    for j in PAR:
        print('par[{}] = {}'.format(Y,j))
        Y += 1
